### ISSUE TITLE
azurerm_data_protection_backup_vault doesn't support Zone Redundancy

### ISSUE THREAD

**Issue by PlanetWilson** [hashicorp/terraform-provider-azurerm#21300]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

When creating a Backup Vault, one should be able to choose Zone Redundancy in addition to Local and Global. 

```
resource "azurerm_data_protection_backup_vault" "bv_test" {
  name                = "bv-test-uksouth-001"
  resource_group_name = "rg-backup"
  location            = "uksouth"
  datastore_type      = "VaultStore"
  redundancy          = "ZoneRedundant"
}
```
Results in :-

```
│ Error: expected redundancy to be one of [GeoRedundant LocallyRedundant], got ZoneRedundant
│ 
│   with azurerm_data_protection_backup_vault.bv_test,
│   on demoenv.tf line 93, in resource "azurerm_data_protection_backup_vault" "bv_test":
│   93:   redundancy          = "ZoneRedundant"

```

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.4.3

### AzureRM Provider Version

3.50.0

### Affected Resource(s)/Data Source(s)

azurerm_data_protection_backup_vault

### Terraform Configuration Files

```hcl
resource "azurerm_data_protection_backup_vault" "bv_test" {
  name                = "bv-test-uksouth-001"
  resource_group_name = "rg-backup"
  location            = "uksouth"
  datastore_type      = "VaultStore"
  redundancy          = "ZoneRedundant"
}
```

### Debug Output/Panic Output

```shell
│ Error: expected redundancy to be one of [GeoRedundant LocallyRedundant], got ZoneRedundant
│ 
│   with azurerm_data_protection_backup_vault.bv_test,
│   on demoenv.tf line 93, in resource "azurerm_data_protection_backup_vault" "bv_test":
│   93:   redundancy          = "ZoneRedundant"
```

### Expected Behaviour

Should create a backup vault with zone redundancy

### Actual Behaviour

Failed to create anything

### Steps to Reproduce

terraform apply on sample provided

**Comment by PlanetWilson** [comment]
I have made a change and [PR](https://github.com/hashicorp/terraform-provider-azurerm/pull/21304) to address this.

**Comment by rcskosir** [comment]
The version upgrade that was needed before this enhancement could be made was merged in #24143.

**Comment by sinbai** [comment]
Hi @PlanetWilson I have submitted PR https://github.com/hashicorp/terraform-provider-azurerm/pull/24556 to support ZoneRedundancy  for azurerm_data_protection_backup_vault. Could you please track it for more updates?

**Comment by tombuildsstuff** [comment]
Fixed via #24556
